### PR TITLE
[Spree 2 Upgrade] WIP - Added adapted order.ship_method=

### LINKED ISSUE #2662: [Spree Upgrade] Replace Order#shipping_method=
`Order#shipping_method=` is gone in Spree 2.0, so any line of code that uses it in OFN will fail. We can provide the same behaviour with an adapter to spree 2 models.

Related to #2009 that handles the absence of order.shipping_method

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #2662

#### What should we test?
Spree upgrade issues cannot be tested yet.

#### How is this related to the Spree upgrade?
Part of order multi-shipments epic.

#### Dependencies
Related to #2654

### CODE REVIEW

File: app/models/spree/order_decorator.rb [hunk 1]
```diff
@@ -333,6 +333,18 @@ def pending_payments
     payments.select {|p| p.state == "checkout"} # Original definition
   end
 
+  # Although Spree 2 supports multi shipments per order, in OFN we keep the rule one shipment per order
+  #   Thus, this method sets the given shipping_method on the first and only shipment in the order
+  def shipping_method=(shipping_method = nil)
+    if shipments.empty?
+      shipment = Spree::Shipment.new
+      shipment.order = self
+      shipment.save
```
Comment (sauloperez) [inline app/models/spree/order_decorator.rb]: AFAIK we don't need this save as AR persists the order and all its associations when doing a `order.save`. Correct me if I'm wrong.

File: app/models/spree/order_decorator.rb [hunk 2]
```diff
@@ -333,6 +333,18 @@ def pending_payments
...
+      shipments << shipment
+    end
+    shipments.first.add_shipping_method(shipping_method, true)
+  end
```
Comment (sauloperez) [inline app/models/spree/order_decorator.rb]: Have we investigated how this works with the way Spree creates shipments in https://github.com/openfoodfoundation/spree/blob/fe0a1311abb097bf3ac8001a24246c6cff5ecdbb/core/app/models/spree/order.rb#L506-L515 ? That's the only place I know in Spree's codebase where shipments are created.

Comment (luisramos0) [reply app/models/spree/order_decorator.rb]: yes, I have seen that code. I am closing this PR actually, we need to fix the specs and then we will see if this is still required.

**Comment by luisramos0** [comment]
I am closing this PR for now, I dont think we will need this adapter. We will have to replace all places where shipping_method= is used. Mostly specs as listed in #2662

**Pull request closed without merging**